uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js

```diff
@@ -919,21 +919,54 @@ define([
             });
 
             describe('when set times are outside the allowed range', function () {
-              var minTime = '09:00';
-              var maxTime = '17:00';
+              initSetTimeAccordingToWorkPattern('02:00', '23:00', '09:00', '17:00');
```

Comment:
We need to avoid functions with 4 parameters as much as possible because it's harder to know what is what. I'm guessing these are the values (before reading the function, correct me if I'm wrong):

- start time for the first day
- end time for the first day
- start time for the last day
- end time for the last day

I'd suggest either of these options:

```js
initSetTimeAccordingToWorkPattern({
  firstDayStartTime: '02:00',
  firstDayEndTime: '23:00',
  lastDayStartTime: '09:00',
  lastDayEndTime: '17:00'
});

// or

initSetTimeForFirstDay('02:00', '23:00');
initSetTimeForLastDay('09:00', '17:00');

```

After reading `initSetTimeAccordingToWorkPattern` I can't understand the name or what it does. This is what I can infer:
* it sets the start and end time for the leave request (nothing to do with Work Pattern then)
* the last two parameters are the response the work pattern will return for the `getWorkDayForDate` method (so not the last day of the work pattern as I suspected).

Why not refactor into two functions then?

```js
setRequestTimes('02:00', '23:00');
setupWorkPatternTimeResponse('09:00', '17:00'); // or a better name
```

`setRequestTimes` would just do that and `setupWorkPatternTimeResponse` does all the `beforeEach`/`afterEach` we need for setting up the tests.

Reply:
I don't really think it deserves two function, this is just to avoid massive code duplication. So I changed the params and name, but left it as one function.